### ISSUE TITLE
BUG: arch-syscall-validate generates misleading diff

### ISSUE THREAD

**Issue by drakenclimber** [seccomp/libseccomp#178]
My system that can reproduce this:
```
$ uname -a
Linux OracleM910 3.10.0-957.12.2.el7.x86_64 #1 SMP Tue May 14 17:35:45 PDT 2019 x86_64 x86_64 x86_64 GNU/Linux
```

Here are my rough steps:
```
cd ~/git/clean
git clone //git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git upstream-torvalds
cd upstream-torvalds
git checkout v5.4-rc4

cd ~/git
git clone https://github.com/pcmoore/misc-libseccomp.git
cd misc-libseccomp
./autogen.sh
./configure
make check-build
cd src/
./arch-syscall-validate ~/git/clean/upstream-torvalds
```

And here's the diff generated by the above `arch-syscall-validate` call:
```
$ ./arch-syscall-validate ~/git/clean/upstream-torvalds
--- mips [library]
+++ mips [system]
@@ -1,3 +1,13 @@
+  SYSTBL  arch/mips/include/generated/asm/syscall_table_32_o32.h
+  SYSTBL  arch/mips/include/generated/asm/syscall_table_64_n32.h
+  SYSTBL  arch/mips/include/generated/asm/syscall_table_64_n64.h
+  SYSTBL  arch/mips/include/generated/asm/syscall_table_64_o32.h
+  SYSHDR  arch/mips/include/generated/uapi/asm/unistd_n32.h
+  SYSHDR  arch/mips/include/generated/uapi/asm/unistd_n64.h
+  SYSHDR  arch/mips/include/generated/uapi/asm/unistd_o32.h
+  SYSNR  arch/mips/include/generated/uapi/asm/unistd_nr_n32.h
+  SYSNR  arch/mips/include/generated/uapi/asm/unistd_nr_n64.h
+  SYSNR  arch/mips/include/generated/uapi/asm/unistd_nr_o32.h
 accept	168
 accept4	334
 access	33
```

**Comment by drakenclimber** [comment]
Interestingly, a subsequent call to `arch-syscall-validate` then works with no diff generated.

But I was able to reproduce the issue by running `git clean -qdfx` and `git reset --hard` on my Linux kernel source repo (`~/git/clean/upstream-torvalds`).

**Comment by drakenclimber** [comment]
I was able to reproduce it on a Fedora30 system as well.  The critical step is having run `git clean -qdfx` and `git reset --hard` on the Linux kernel source directory.

**Comment by pcmoore** [comment]
Testing this again just now, it appears that x32, mips, mips64, and mips64n32 are the worst with parisc64 slightly affected.

**Comment by pcmoore** [comment]
> Testing this again just now, it appears that x32, mips, mips64, and mips64n32 are the worst with parisc64 slightly affected.

Hmm, nevermind, it looks like there are still some issues with the arch-syscall-validate script.

**Comment by pcmoore** [comment]
... and parisc/parisc64 isn't quite right.  The good news is that it would appear this particular bug is now gone :)

Once I fix the other stuff I'll try the reproducer again, but I think we might be able to close out this issue.

**Comment by pcmoore** [comment]
This is still a problem, but I believe I have a fix (essentially move the MIPS ABIs over to the new syscall.tbl format).

**Comment by pcmoore** [comment]
This should be fixed in afc713a8fe289059a18350ab550ca28056356196.
